drivers/sensor/bmi08x/bmi08x_accel.c

```diff
@@ -0,0 +1,775 @@
+
+static int bmi08x_accel_transceive_spi(const struct device *dev, uint8_t reg, bool write,
+				       void *data, size_t length)
+{
+	const struct bmi08x_accel_config *bmi08x = dev->config;
+	const struct spi_buf tx_buf[2] = {{.buf = &reg, .len = 1}, {.buf = data, .len = length}};
+	const struct spi_buf_set tx = {.buffers = tx_buf, .count = write ? 2 : 1};
+
+	if (!write) {
+		uint16_t dummy;
+		const struct spi_buf rx_buf[2] = {{.buf = &dummy, .len = 2},
+						  {.buf = data, .len = length}};
+		const struct spi_buf_set rx = {.buffers = rx_buf, .count = 2};
+
+		return spi_transceive_dt(&bmi08x->bus.spi, &tx, &rx);
+	}
+
+	return spi_write_dt(&bmi08x->bus.spi, &tx);
+}
+
+#if BMI08X_ACCEL_ANY_INST_HAS_DATA_SYNC
+static int bmi08x_write_config_file_spi(const struct device *dev)
+{
+	if (bmi08x_accel_byte_write(dev, BMI08X_ACCEL_RESERVED_5B_REG, 0) < 0) {
+		LOG_ERR("Cannot write index");
+		return -EIO;
+	}
+	if (bmi08x_accel_byte_write(dev, BMI08X_ACCEL_RESERVED_5C_REG, 0) < 0) {
+		LOG_ERR("Cannot write index");
+		return -EIO;
+	}
+	/* write config file */
+	if (bmi08x_accel_write(dev, BMI08X_ACCEL_FEATURE_CFG_REG, (uint8_t *)bmi08x_config_file,
+			       sizeof(bmi08x_config_file))) {
+		LOG_ERR("Cannot write configuration for accelerometer.");
+		return -EIO;
+	}
+	return 0;
+}
+#endif
+
+static int bmi08x_bus_check_spi(const union bmi08x_bus *bus)
+{
+	return spi_is_ready_dt(&bus->spi) ? 0 : -ENODEV;
+}
+
+static int bmi08x_bus_init_spi(const struct device *dev)
+{
+	uint8_t val;
+	/* do a dummy read from 0x7F to activate SPI */
+	if (bmi08x_accel_byte_read(dev, 0x7F, &val) < 0) {
+		LOG_ERR("Cannot read from 0x7F..");
+		return -EIO;
+	}
+
+	k_usleep(100);
+
+	return 0;
+}
+
+static const struct bmi08x_accel_bus_io bmi08x_spi_api = {.check = bmi08x_bus_check_spi,
+							  .bus_init = bmi08x_bus_init_spi,
+							  .transceive = bmi08x_accel_transceive_spi,
+#if BMI08X_ACCEL_ANY_INST_HAS_DATA_SYNC
+							  .write_config_file =
+								  bmi08x_write_config_file_spi
+#endif
+};
+
+#endif /* DT_ANY_INST_ON_BUS_STATUS_OKAY(spi) */
+
+static inline int bmi08x_bus_check(const struct device *dev)
+{
+	const struct bmi08x_accel_config *config = dev->config;
+
+	return config->api->check(&config->bus);
+}
+
+static inline int bmi08x_bus_init(const struct device *dev)
+{
+	const struct bmi08x_accel_config *config = dev->config;
+
+	/* optional, only needed to initialize SPI according to datasheet */
+	if (config->api->bus_init) {
+		return config->api->bus_init(dev);
+	}
+	return 0;
+}
+
+static int bmi08x_accel_transceive(const struct device *dev, uint8_t reg, bool write, void *data,
+				   size_t length)
+{
+	const struct bmi08x_accel_config *config = dev->config;
+
+	return config->api->transceive(dev, reg, write, data, length);
+}
+
+int bmi08x_accel_read(const struct device *dev, uint8_t reg_addr, uint8_t *data, uint8_t len)
+{
+	return bmi08x_accel_transceive(dev, reg_addr | BIT(7), false, data, len);
```

Comment:
reg_addr | BIT(7)

In spi or i2c the address operate is different, I think reg address can not operate here.

Reply:
I do not understand what you are trying to say here, but BIT(7) Must be set to indicate a read of the reg